pkg/mapper/examples/batchmap/main.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/numaproj/numaflow-go/pkg/mapper"
+)
+
+func mapFn(_ context.Context, datums []mapper.Datum) mapper.BatchResponses {
+	batchResponses := mapper.BatchResponsesBuilder()
+	log.Println("MYDEBUG: length of input ", len(datums))
+	for _, d := range datums {
+		msg := d.Value()
+		_ = d.EventTime() // Event time is available
+		_ = d.Watermark() // Watermark is available
+		results := mapper.NewBatchResponse(d.Id())
```

Comment:
Is an empty response valid to ensure prop ACK'ing of message without necessarily assigning a response message to it? 

Ex. my batch is a list of JSON, but in my handler I take combine them into a list of JSON (psedo-reduce). One of the `NewBatchResponse` would align with the new message, but the rest I would want to acknowledge as complete, but they wouldn't have an actual response.

Reply:
We would still need to add an empty batch response for those IDs with an empty list. We expect that the number of requests in batch and number of responses to be equal. 
But for you use case if you pass on an empty response with the correct ID, those should be acked 
`
results := mapper.NewBatchResponse(d.Id())
batchResponses = batchResponses.Append(results)
`